### ISSUE TITLE
PredTransform() for Python

### ISSUE THREAD

**Issue by Roffild** [dmlc/xgboost#6445]
```python
# NOTE: when you do customized loss function, the default prediction value is
# margin, which means the prediction is score before logistic transformation.
def evalerror(preds, dtrain):
    labels = dtrain.get_label()
    preds = 1.0 / (1.0 + np.exp(-preds))  # transform raw leaf weight
    # return a pair metric_name, result. The metric name must not contain a
    # colon (:) or a space
    return 'my-error', float(sum(labels != (preds > 0.5))) / len(labels)
```
We need [PredTransform()](https://github.com/dmlc/xgboost/blob/927c316aeb0a948e9635c1ffe54d5e6ed25f6797/src/objective/regression_obj.cu#L235) so as not to depend on different types of "objective".

**Comment by trivialfis** [comment]
In order to avoiding depending on objective, the output prediction for customized loss function is untransformed prediction (leave value).  Why do you need `PredTransform`?

**Comment by Roffild** [comment]
All standard metrics are called after PredTransform().
https://github.com/dmlc/xgboost/blob/927c316aeb0a948e9635c1ffe54d5e6ed25f6797/src/learner.cc#L1086-L1090

Therefore, no transformation is needed.
https://github.com/dmlc/xgboost/blob/927c316aeb0a948e9635c1ffe54d5e6ed25f6797/src/metric/elementwise_metric.cu#L130-L142

**Comment by trivialfis** [comment]
Let me summarize your feature request:

You want to implement a evaluation metric that's orthogonal to objective functions.  To do that, you want the transformed prediction output so that your evaluation metric doesn't have to do the transformation.

If that's a fair summary, I guess we can have some kind of additional parameter or object method informing xgboost to do the transformation.

**Comment by trivialfis** [comment]
I changed the title of this issue.  Feel free to change it again if it's not an accurate description.

**Comment by Roffild** [comment]
The objective function type is not passed to my metric evaluation function. Therefore, even having crammed all types of transformations (regression, multi, etc.), choosing will not work.
